### PR TITLE
chore: merge release/v1.20.0 into master

### PR DESCRIPTION
**Pull request by arajasek**
## Related Issues

We will be continually performing such merges so that the latest FEVM work (which is landing in `release/v1.20.0` also lands in master.

## Additional Info

Review note: This has already been reviewed and merged in #9998. I think review of this PR should focus on merge issues that may arise from the refactor in #9481, with particular attention being paid to the `chain/consensus` code.

## Checklist

Before you mark the PR ready for review, please make sure that:

- [ ] Commits have a clear commit message.
- [ ] PR title is in the form of of `<PR type>: <area>: <change being made>`
  - example: ` fix: mempool: Introduce a cache for valid signatures`
  - `PR type`: fix, feat, build, chore, ci, docs, perf, refactor, revert, style, test
  - `area`, e.g. api, chain, state, market, mempool, multisig, networking, paych, proving, sealing, wallet, deps
- [ ] New features have usage guidelines and / or documentation updates in
  - [ ] [Lotus Documentation](https://lotus.filecoin.io)
  - [ ] [Discussion Tutorials](https://github.com/filecoin-project/lotus/discussions/categories/tutorials)
- [ ] Tests exist for new functionality or change in behavior
- [ ] CI is green

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -84,6 +84,12 @@ butterflynet: build-devnets
 interopnet: GOFLAGS+=-tags=interopnet
 interopnet: build-devnets
 
+wallabynet: GOFLAGS+=-tags=wallabynet
+wallabynet: build-devnets
+
+hyperspacenet: GOFLAGS+=-tags=hyperspacenet
+hyperspacenet: build-devnets
```
Comment (magik6k) [inline Makefile:91]: Drop those?

File: chain/messagepool/messagepool.go [hunk 2]
```diff
@@ -853,18 +851,32 @@ func (mp *MessagePool) addTs(ctx context.Context, m *types.SignedMessage, curTs
...
+	senderAct, err := mp.api.GetActorAfter(m.Message.From, curTs)
+	if err != nil {
+		return false, xerrors.Errorf("failed to get sender actor: %w", err)
+	}
+
+	// This message can only be included in the _next_ epoch and beyond, hence the +1.
+	epoch := curTs.Height() + 1
+	nv := mp.api.StateNetworkVersion(ctx, epoch)
+
+	// TODO: I'm not thrilled about depending on filcns here, but I prefer this to duplicating logic
```
Comment (magik6k) [inline chain/messagepool/messagepool.go:863]: I guess this now depends on the main consensus package, which is more fine?

File: chain/types/tipset.go [hunk 3]
```diff
@@ -196,8 +196,23 @@ func (ts *TipSet) MinTicket() *Ticket {
...
+	blks := ts.Blocks()
+
+	// TODO::FVM @vyzo @magik Null rounds shouldn't ever be represented as
+	//  tipsets with no blocks; Null-round generally means that the tipset at
+	//  that epoch doesn't exist - and the next tipset that does exist links
+	//  straight to first epoch with blocks (@raulk agrees -- this is odd)
+	if len(blks) == 0 {
+		// null rounds make things crash -- it is threaded in every fvm instantiation
+		return 0
+	}
```
Comment (magik6k) [inline chain/types/tipset.go:212]: Did we ever get to why this is/was needed? Can we drop this?

File: chain/vm/gas.go [hunk 4]
```diff
@@ -111,6 +111,7 @@ var Prices = map[abi.ChainEpoch]Pricelist{
 		verifySignature: map[crypto.SigType]int64{
 			crypto.SigTypeBLS:       16598605,
 			crypto.SigTypeSecp256k1: 1637292,
+			crypto.SigTypeDelegated: 1637292,
```
Comment (magik6k) [inline chain/vm/gas.go:114]: Don't we also need this in pricelists for other versions?

File: itests/splitstore_test.go [hunk 5]
```diff
@@ -307,6 +350,16 @@ func splitStorePruneIndex(ctx context.Context, t *testing.T, n *kit.TestFullNode
 	return pruneIndex
 }
 
+func ipldExists(ctx context.Context, t *testing.T, c cid.Cid, n *kit.TestFullNode) bool {
+	_, err := n.ChainReadObj(ctx, c)
```
Comment (magik6k) [inline itests/splitstore_test.go:354]: nit: There is `ChainHasObj`

File: node/impl/full/chain.go [hunk 6]
```diff
@@ -654,6 +656,33 @@ func (a *ChainAPI) ChainBlockstoreInfo(ctx context.Context) (map[string]interfac
 	return info.Info(), nil
 }
 
+// ChainGetEvents returns the events under an event AMT root CID.
+//
+// TODO (raulk) make copies of this logic elsewhere use this (e.g. itests, CLI, events filter).
```
Comment (magik6k) [inline node/impl/full/chain.go:661]: todo

File: node/impl/full/dummy.go [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+	"context"
+	"errors"
+
+	"github.com/filecoin-project/lotus/api"
+	"github.com/filecoin-project/lotus/chain/types/ethtypes"
+)
+
+var ErrImplementMe = errors.New("Not implemented yet")
+
+type EthModuleDummy struct{}
```
Comment (magik6k) [inline node/impl/full/dummy.go:13]: This should be removed (it was removed in the eth_ gateway PR, not sure why it ressurected itself, but probably when merge conflicts were getting fixed)

**Review by magik6k — APPROVED** [review]

File: build/actors/pack.sh [hunk 8]
```diff
@@ -1,6 +1,6 @@
 #!/bin/bash
 
-NETWORKS=(devnet mainnet caterpillarnet butterflynet testing testing-fake-proofs calibrationnet)
+NETWORKS=(devnet mainnet caterpillarnet butterflynet testing testing-fake-proofs calibrationnet hyperspace)
```
[the remark this answers is not in the document]
Comment (geoff-vball) [reply build/actors/pack.sh:3]: Yes this file generates `build/builtin_actors_gen.go`. I'm not sure about Wallaby

**Comment by magik6k** [comment]
(meant to comment, not approve)

File: chain/vm/gas.go (hunk 4 above)
Comment (arajasek) [reply chain/vm/gas.go:114]: No, nothing should ever call it anyway?

File: node/impl/full/dummy.go (hunk 7 above)
Comment (geoff-vball) [reply node/impl/full/dummy.go:13]: We're using it again since this PR https://github.com/filecoin-project/lotus/pull/10072

**Review by geoff-vball — APPROVED** [review]

**Pull request merged**